signer/src/storage/model.rs

```diff
@@ -1195,6 +1195,120 @@ pub struct BitcoinWithdrawalOutput {
+    pub signer_bitmap: BitArray<[u8; 16]>,
+    /// This is the outpoint for the bitcoin transaction that serviced the
+    /// request.
+    pub outpoint: OutPoint,
+    /// This is the fee that was spent to the bitcoin miners to confirm the
+    /// withdrawal request.
+    pub fee: u64,
+    /// The bitcoin block hash where the sweep transaction was included.
+    pub sweep_block_hash: BitcoinBlockHash,
+    /// The bitcoin block height where the sweep transaction was included.
+    pub sweep_block_height: u64,
+    /// The transaction id of the bitcoin transaction that fulfilled the
+    /// withdrawal request.
+    pub sweep_txid: BitcoinTxid,
+}
+
+impl From<sbtc::events::WithdrawalAcceptEvent> for WithdrawalAcceptEvent {
+    fn from(sbtc_event: sbtc::events::WithdrawalAcceptEvent) -> WithdrawalAcceptEvent {
+        let mut reversed_raw_hash = *sbtc_event.sweep_block_hash.as_bytes();
+        reversed_raw_hash.reverse();
+        let raw_hash = Hash::from_byte_array(reversed_raw_hash);
+        let sweep_hash = BitcoinBlockHash::from_raw_hash(raw_hash);
+        let txid = StacksTxid::from(sbtc_event.txid.0);
+        WithdrawalAcceptEvent {
+            txid,
+            block_id: sbtc_event.block_id,
+            request_id: sbtc_event.request_id,
+            signer_bitmap: BitArray::new(sbtc_event.signer_bitmap.to_le_bytes()),
+            outpoint: sbtc_event.outpoint,
+            fee: sbtc_event.fee,
+            sweep_block_hash: sweep_hash,
+            sweep_block_height: sbtc_event.sweep_block_height,
+            sweep_txid: sbtc_event.sweep_txid,
+        }
+    }
+}
+
+
+/// This is the event that is emitted from the `create-withdrawal-request`
+/// public function in sbtc-registry smart contract.
+#[derive(Debug, Clone)]
+pub struct CompletedDepositEvent {
+    /// The transaction id of the stacks transaction that generated this
+    /// event.
+    pub txid: StacksTxid,
+    /// The block ID of the block for this event.
+    pub block_id: StacksBlockId,
+    /// This is the amount of sBTC to mint to the intended recipient.
+    pub amount: u64,
+    /// This is the outpoint of the original bitcoin deposit transaction.
+    pub outpoint: OutPoint,
+    /// The bitcoin block hash where the sweep transaction was included.
+    pub sweep_block_hash: BitcoinBlockHash,
+    /// The bitcoin block height where the sweep transaction was included.
+    pub sweep_block_height: u64,
+    /// The transaction id of the bitcoin transaction that fulfilled the
+    /// deposit.
+    pub sweep_txid: BitcoinTxid,
+}
+
+impl From<sbtc::events::CompletedDepositEvent> for CompletedDepositEvent {
+    fn from(sbtc_event: sbtc::events::CompletedDepositEvent) -> CompletedDepositEvent {
+        let mut reversed_raw_hash = *sbtc_event.sweep_block_hash.as_bytes();
+        reversed_raw_hash.reverse();
+        let raw_hash = Hash::from_byte_array(reversed_raw_hash);
+        let sweep_hash = BitcoinBlockHash::from_raw_hash(raw_hash);
```

Comment:
Same here, you can replace this with:
```suggestion
        let sweep_hash = BitcoinBlockHash::from(sbtc_event.sweep_block_hash);
```